lib/rules/prefer-presence-queries.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const { getDocsUrl, ALL_QUERIES_METHODS } = require('../utils');
+
+const QUERIES_REGEXP = new RegExp(
+  `^(get|query)(All)?(${ALL_QUERIES_METHODS.join('|')})$`
+);
+const PRESENCE_MATCHERS = ['toBeInTheDocument', 'toBeTruthy', 'toBeDefined'];
+const ABSENCE_MATCHERS = ['toBeNull', 'toBeFalsy'];
+
+module.exports = {
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description:
+        'Ensure appropriate get*/query* queries are used with their respective matchers',
+      recommended: 'error',
+      url: getDocsUrl('prefer-presence-queries'),
+    },
+    messages: {
+      presenceQuery:
+        'Use `getBy*` queries rather than `queryBy*` for checking element is present',
+      absenceQuery:
+        'Use `queryBy*` queries rather than `getBy*` for checking element is NOT present',
+      expectQueryBy:
+        'Use `getBy*` only when checking elements are present, otherwise use `queryBy*`',
+    },
+    schema: [],
+    type: 'suggestion',
+    fixable: null,
+  },
+
+  create: context => ({
+    [`CallExpression Identifier[name=${QUERIES_REGEXP}]`](node) {
+      const expectCallNode = findClosestCallNode(node, 'expect');
+
+      if (expectCallNode) {
+        const expectStatement = expectCallNode.parent;
+        let matcher = expectStatement.property.name;
+        let isNegatedMatcher = false;
+
+        if (matcher === 'not') {
+          isNegatedMatcher = true;
+          matcher =
+            expectStatement.parent &&
```

Comment:
I think this can removed? On line 39, there is the same assignment.
If these properties can be `undefined` perhaps it would be safer to rewrite line 39, the same way like this assignment.

Reply:
It's not the same assignment tho. On line 39 is getting property name of expect statement. Here is getting property name from **parent** of expect statement. This would be clearer if getting the matcher with some util similar to `findClosestCallNode`.

You are right about guarding against nil on both cases.

Reply:
Oh shoot you're right.
I was so sure because I read it twice, sorry for the confussion 😅

You could create a util function, but I'm not sure if it would add the value as `findClosestCallNode` does